### PR TITLE
[Bug 1769109] Fix clients_last_seen_joined schedule

### PR DESCRIPTION
**Pull request by scholtzan**
Depends on https://github.com/mozilla/bigquery-etl/pull/3069

### CODE REVIEW

File: dags/copy_deduplicate.py [hunk 1]
```diff
@@ -380,13 +380,33 @@
             "query",
             "backfill",
             "*.clients_last_seen_joined_v1",
-        ] + baseline_args,
+        ] + [    # metrics pings are usually delayed by one day after their logical activity period
+            "--project-id=moz-fx-data-shared-prod",
+            "--start_date={{ macros.ds_add(ds, -1) }}",
+            "--end_date={{ macros.ds_add(ds, -1) }}",
```
Comment (kik-kik) [inline dags/copy_deduplicate.py:386]: are the start and end date values meant to be the same?

File: dags/copy_deduplicate.py [hunk 2]
```diff
@@ -380,13 +380,33 @@
...
+
+        baseline_clients_last_seen >> baseline_clients_last_seen_external
 
     telemetry_derived__core_clients_first_seen__v1 >> baseline_clients_first_seen
     baseline_clients_first_seen >> baseline_clients_daily
     copy_deduplicate_all >> baseline_clients_daily >> baseline_clients_last_seen
     copy_deduplicate_all >> metrics_clients_daily >> metrics_clients_last_seen
     metrics_clients_last_seen >> clients_last_seen_joined
     baseline_clients_last_seen >> clients_last_seen_joined
+    clients_last_seen_joined >> clients_last_seen_joined_external
```
Comment (sean-rose) [inline dags/copy_deduplicate.py:412]:
```suggestion
        clients_last_seen_joined >> clients_last_seen_joined_external

    telemetry_derived__core_clients_first_seen__v1 >> baseline_clients_first_seen
    baseline_clients_first_seen >> baseline_clients_daily
    copy_deduplicate_all >> baseline_clients_daily >> baseline_clients_last_seen
    copy_deduplicate_all >> metrics_clients_daily >> metrics_clients_last_seen
    metrics_clients_last_seen >> clients_last_seen_joined
    baseline_clients_last_seen >> clients_last_seen_joined
```

File: dags/copy_deduplicate.py [hunk 3]
```diff
@@ -380,13 +380,33 @@
...
+            ("bqetl_unified", "hour=3, minute=0"),
+        }
+
+        for downstream_dependency in downstream_dependencies:
+            ExternalTaskMarker(
+                task_id=f"{downstream_dependency[0]}__wait_for_clients_last_seen_joined",
+                external_dag_id=downstream_dependency[0],
+                external_task_id="wait_for_clients_last_seen_joined",
+                execution_date="{{ execution_date.replace(" + downstream_dependency[1] + ").isoformat() }}",
+            )
```
Comment (sean-rose) [inline dags/copy_deduplicate.py:402]: An idea to make the variables more self-explanatory:
```suggestion
        downstream_dag_times = {
            "bqetl_unified": "hour=3, minute=0",
        }

        for downstream_dag, dag_time_parts in downstream_dag_times.items():
            ExternalTaskMarker(
                task_id=f"{downstream_dag}__wait_for_clients_last_seen_joined",
                external_dag_id=downstream_dependency[0],
                external_task_id="wait_for_clients_last_seen_joined",
                execution_date="{{ execution_date.replace(" + dag_time_parts + ").isoformat() }}",
            )
```

**Review by sean-rose — APPROVED** [review]
r+wc

File: dags/copy_deduplicate.py (hunk 3 above)
Comment (scholtzan) [reply dags/copy_deduplicate.py]: Indeed, that looks a little cleaner 👍

File: dags/utils/gcp.py [hunk 4]
```diff
@@ -221,7 +221,8 @@ def bigquery_etl_query(
     kwargs["name"] = kwargs.get("name", kwargs["task_id"].replace("_", "-"))
     if not project_id:
         project_id = "moz-fx-data-shared-prod"
-    sql_file_path = sql_file_path or "sql/{}/{}/{}/query.sql".format(project_id, dataset_id, destination_table)
+    destination_table_no_partition = destination_table.split("$")[0]
+    sql_file_path = sql_file_path or "sql/{}/{}/{}/query.sql".format(project_id, dataset_id, destination_table_no_partition)
```
Comment (scholtzan) [inline dags/utils/gcp.py:225]: This should address https://github.com/mozilla/bigquery-etl/pull/3069/files/443cd7857e6c55fdbc0214bc15a315f073a2cacd#r909105233

**Pull request merged**